freqtrade/freqai/freqai_interface.py

```diff
@@ -181,29 +183,36 @@ def _start_scanning(self, strategy: IStrategy) -> None:
         """
         while not self._stop_event.is_set():
             time.sleep(1)
-            for pair in self.config.get("exchange", {}).get("pair_whitelist"):
-
-                (_, trained_timestamp, _) = self.dd.get_pair_dict_info(pair)
-
-                if self.dd.pair_dict[pair]["priority"] != 1:
-                    continue
-                dk = FreqaiDataKitchen(self.config, self.live, pair)
-                dk.set_paths(pair, trained_timestamp)
-                (
-                    retrain,
-                    new_trained_timerange,
-                    data_load_timerange,
-                ) = dk.check_if_new_training_required(trained_timestamp)
-                dk.set_paths(pair, new_trained_timerange.stopts)
-
-                if retrain:
-                    self.train_timer('start')
-                    self.extract_data_and_train_model(
-                        new_trained_timerange, pair, strategy, dk, data_load_timerange
-                    )
-                    self.train_timer('stop')
-
-            self.dd.save_historic_predictions_to_disk()
+            pair = self.train_queue[0]
+
+            # ensure pair is avaialble in dp
+            if pair not in strategy.dp.current_whitelist():
+                self.train_queue.popleft()
+                logger.warning(f'{pair} not in current whitelist, removing from train queue.')
```

Comment:
I guess we should also be able to do this in reverse? (maybe not on every iteration)
so basically - we check for each pair in the current pairlist, if it's in the train-queue - and if it is not - we append it at the end?

Now it's an idea for the future for sure - as that'll also have implications on the data - which might be missing for that new pair.

Reply:
Sure it just needs to also execute the refresh ohlcv. There is your volume pairlist support if you want it!

Reply:
absolutely, that's my point 
needs to be looked at separately though.